static/scripts/app/views/dir.js

```diff
@@ -50,6 +51,45 @@ define([
                         tNext: gettext("Next (Right arrow key)"), // Alt text on right arrow
                         tCounter: gettext("%curr% of %total%") // Markup for "1 of 7" counter
                     },
+                    callbacks: {
+                        elementParse: function(item) {
+                            var repo_id = _this.dir.repo_id,
+                                cur_path = _this.dir.path,
+                                cur_img = _this.dir.find(function(dirent) {
+                                    return dirent.get('obj_name') == item.el[0].innerText;
```

Comment:
Is this way okay to retrieve the file entity from the current directory? Just by having the <a> element I couldn't get a better way as using the filename as seach parameter. A better way would be to store the obj_name as data-filename attribute in the html code during directory generation.

Okay? or do you've a better solution.